File: lib/devUtils/resetMockDB.ts
import fs from 'fs'
import path from 'path'

/* // Path to the mock in-memory directory
const directory = new URL('.', import.meta.url).pathname
const __dirname = process.platform === 'win32' ? directory.slice(1) : directory
console.log('__dirname: ' + __dirname)
const mockDir = path.join(__dirname, '../../mock/inMemory')
console.log('mockDir: ' + mockDir)

// Get all files in the mock in-memory directory
const files = fs.readdirSync(mockDir)

// Object to store the final data
const db = {} */

/* // Process each file
// for (const file of files) {
const file = files[0]
// Only process TypeScript files
if (path.extname(file) === '.ts') {
  // Dynamically import the module
  console.log('file: ' + file)
  const filePath = path.join(mockDir, file)
  console.log('filePath: ' + filePath)
  // Extract the entity name from the file name
  const entity = path.basename(file, '.ts')
  console.log('entity: ' + entity)
  const myObject = await loadObject(filePath, entity)
  console.log('myObject: ' + myObject)
  // import(filePath).then((module) => console.log('module: ' + module))
  // Extract the raw data array
  // Assumes the raw data is named as "<entity>Datas" (e.g., accountDatas)
  // const rawDataArrayName = `${entity}Datas`
  // loadObject(filePath, rawDataArrayName)
  // import(path.join(mockDir, file)).then((module) => {
  //
  //       if (module[rawDataArrayName]) {
  //         db[entity] = module[rawDataArrayName]
  //       }
  //
  //       // Check if we have processed all files
  //       if (Object.keys(db).length === files.length) {
  //         // Write to db.json
  //         const outputPath = path.join(__dirname, '../../mock/jsonServer/new_db.json')
  //         fs.writeFileSync(outputPath, JSON.stringify(db, null, 2), 'utf-8')
  //       }
  // })
  // }
}

function print(obj) {
  for (const [key, value] of Object.entries(obj)) {
    console.log(`${key}: Value: ${value}`)
  }
}

 */

export async function loadObject(fileName: string, objectName: string) {
  const module = await import(fileName)
  const myObject = module[objectName] // use the dynamic name here
  return myObject
}

const accountDatas = await loadObject('../../mock/inMemory/account.ts', 'accountDatas')

console.log('accountDatas:', accountDatas) // Logs the object to the console

File: mock/utils/resetMockDB.py
import os
import json

# Determine paths
current_dir = os.path.dirname(os.path.abspath(__file__))
print(f"Current dir: {current_dir}")
mock_dir = os.path.join(current_dir, '..\..\mock\inMemory')
print(f"Mock dir: {mock_dir}")

# Get all files in the mock in-memory directory
files = [f for f in os.listdir(
    mock_dir) if os.path.isfile(os.path.join(mock_dir, f))]
print(f"Files: {files}")
# Object to store the final data
db = {}
#
# Process each file
for file in files:
    # Only process TypeScript files
    if file.endswith('.ts'):
        # Extract the entity name from the file name
        entity = os.path.splitext(file)[0]

        # Assumes the raw data is named as "<entity>Datas" (e.g., accountDatas)
        raw_data_array_name = f"{entity}Datas"
        print(f"Entity: {entity}, raw data array name: {raw_data_array_name}")

        # Dynamically import the module
        module = {}
        with open(os.path.join(mock_dir, file), 'r') as f:
            # This is a naive way to convert TypeScript exports to Python dicts.
            # Assumes the exports are at the end and in JSON-like format.
            for line in f.readlines():
                if raw_data_array_name in line:
                    data_str = line.split('=')[1].strip().rstrip(';')
                    module[raw_data_array_name] = json.loads(data_str)

        if raw_data_array_name in module:
            db[entity] = module[raw_data_array_name]
            print(f"Entity: {entity}, data: {db[entity]}")
#
# # Write to db.json
# output_path = os.path.join(current_dir, '..\mew_db.json')
# with open(output_path, 'w') as f:
#     json.dump(db, f, indent=2)
